pkg/image/registryclient/alternativesources.go

```diff
@@ -160,62 +146,40 @@ func (c *Context) PreferredImageSource(image imagereference.DockerImageReference
 	var altSources []imagereference.DockerImageReference
 	// error if given an icsp file and don't successfully connect to image repository
 	if len(c.ImageSources.ICSPFile) > 0 {
-		if err = c.AddImageSourcePoliciesFromFile(); err != nil {
+		if err = c.addImageSourcePoliciesFromFile(); err != nil {
 			return imagereference.DockerImageReference{}, nil, nil, err
 		}
-		altSources, err = c.AddImageSources(image)
+		altSources, err = c.addImageSources(image)
 		if err != nil {
 			return imagereference.DockerImageReference{}, nil, nil, err
 		}
 		for _, icsRef := range altSources {
 			replacedImage = replaceImage(icsRef, image)
-			// if not successful, error will be handled below
 			if repo, err = c.Repository(ctx, replacedImage.DockerClientDefaults().RegistryURL(), replacedImage.RepositoryName(), insecure); err == nil {
 				manifests, err = repo.Manifests(ctx)
 				return replacedImage, repo, manifests, err
 			}
 		}
 	}
 
-	// error if explicitly asked to look in cluster (from command line flag) and didn't connect to image repository
-	if c.ImageSources.LookupClusterICSP {
-		// now try to look for ICSPs from cluster
-		if c.ImageSources.ICSPClientFn == nil {
-			return imagereference.DockerImageReference{}, nil, nil, fmt.Errorf("unable to find ImageContentSourcePolicy object from cluster")
-		}
-		if err = c.AddICSPsFromCluster(true); err != nil {
-			return imagereference.DockerImageReference{}, nil, nil, err
-		}
-		altSources, err = c.AddImageSources(image)
-		if err != nil {
-			return imagereference.DockerImageReference{}, nil, nil, err
-		}
-		for _, icsRef := range altSources {
-			replacedImage = replaceImage(icsRef, image)
-			if repo, err = c.Repository(ctx, replacedImage.DockerClientDefaults().RegistryURL(), replacedImage.RepositoryName(), insecure); err == nil {
-				manifests, err = repo.Manifests(ctx)
-				return replacedImage, repo, manifests, err
-			}
-		}
+	// now try to look for ICSPs from cluster
+	icspClient, err := c.ImageSources.ICSPClientFn()
+	if err != nil {
+		return imagereference.DockerImageReference{}, nil, nil, err
 	}
-
-	// try implicit lookup only if alwaysTryAlternativeSources is true
-	if alwaysTryAlternativeSources {
-		if err = c.AddICSPsFromCluster(false); err != nil {
-			return imagereference.DockerImageReference{}, nil, nil, err
-		}
-		altSources, err = c.AddImageSources(image)
-		if err != nil {
-			return imagereference.DockerImageReference{}, nil, nil, err
-		}
-		for _, icsRef := range altSources {
-			replacedImage = replaceImage(icsRef, image)
-			if repo, err = c.Repository(ctx, replacedImage.DockerClientDefaults().RegistryURL(), replacedImage.RepositoryName(), insecure); err == nil {
-				manifests, err = repo.Manifests(ctx)
-				return replacedImage, repo, manifests, err
-			}
+	c.addICSPsFromCluster(icspClient)
+	altSources, err = c.addImageSources(image)
```

Comment:
Now that I'm re-reading this `addImageSources` is more like getter for alternative image sources for the provided image. How about renaming that to `alternateImageSources`?

Reply:
updated